worker-api/src/main/java/com/microfocus/caf/api/worker/QueueTaskMessage.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2015-2021 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.microfocus.caf.api.worker;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.validation.constraints.NotNull;
+
+import com.hpe.caf.api.worker.TaskSourceInfo;
+import com.hpe.caf.api.worker.TaskStatus;
+import com.hpe.caf.api.worker.TrackingInfo;
+
+/**
+ * The generic task message class to be serialised from or to a queue. This will contain the serialised worker-specific data inside.
+ */
+public final class QueueTaskMessage
```

Comment:
Do we really need this new class?  Couldn't the `TaskMessage` class be changed or was that causing too many issues?

Reply:
There are 2 reasons to keep it -
- We would need to hold on to this class atleast till we are clear about the changes for step 2 i.e. worker publishing the new format. 
- TaskMessage has very thick integration with the worker code. I am tempted to keep a separate class which would be only responsible to transfer data within the pipes. But that also can be decided once we are done with step 2.

Reply:
Ok fair enough.